### PR TITLE
Fix receptor work unit release after completion

### PR DESCRIPTION
**Pull request by TheRealHaoLiu**
##### SUMMARY
Fix bug introduced by https://github.com/ansible/awx/pull/15392 that cause workunit to NOT be auto released after job completes

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME

 - API

##### AWX VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Review by AlanCoding — APPROVED** [review]

File: awx/main/tasks/receptor.py [hunk 1]
```diff
@@ -403,13 +403,36 @@ def run(self):
...
-            if self.unit_id is not None and settings.RECEPTOR_RELEASE_WORK:
-                if settings.RECPETOR_KEEP_WORK_ON_ERROR and getattr(res, 'status', 'error') == 'error':
-                    try:
-                        receptor_ctl.simple_command(f"work release {self.unit_id}")
-                    except Exception:
-                        logger.exception(f"Error releasing work unit {self.unit_id}.")
+            status = getattr(res, 'status', 'error')
+            self._receptor_release_work(receptor_ctl, status)
+
+    def _receptor_release_work(self, receptor_ctl: ReceptorControl, status: str) -> None:
```
Comment (chrismeyersfsu) [inline awx/main/tasks/receptor.py:409]: Should we log a message on, basically, every code path taken in this function?

File: awx/main/tasks/receptor.py [hunk 2]
```diff
@@ -403,13 +403,36 @@ def run(self):
...
+        """
+        Releases the work unit from Receptor if certain conditions are met.
+        This method checks several conditions before attempting to release the work unit:
+        - If `self.unit_id` is `None`, the method returns immediately.
+        - If the `RECEPTOR_RELEASE_WORK` setting is `False`, the method returns immediately.
+        - If the `RECEPTOR_KEEP_WORK_ON_ERROR` setting is `True` and the status is 'error', the method returns immediately.
+        If all conditions are met, the method attempts to release the work unit using the Receptor control command.
```
Comment (chrismeyersfsu) [inline awx/main/tasks/receptor.py]: > If all conditions are met

If none of the above conditions are met

File: awx/main/tasks/receptor.py (hunk 1 above)
Comment (TheRealHaoLiu) [reply awx/main/tasks/receptor.py:409]: fine...

**Comment by TheRealHaoLiu** [comment]
@chrismeyersfsu addressed your comments please re-review

**Pull request merged**